portfolio/src/main/webapp/script.js

```diff
@@ -44,6 +44,16 @@ const PREV_SLIDE_DIR = -1;
 const NEXT_SLIDE_DIR = 1;
 let slideIndices = [0, 0, 0]; // array of slide indices for each slide deck
 
+function preloadImages()
+{
+    for(let i = 0; i < SLIDES.length; i++) {
+        for(let j = 0; j < SLIDES[i].length; j++) {
+            var img = new Image();
+            img.src = SLIDES[i][j];
+        }
+    }
```

Comment:
If you want to iterate through an array and not really care about indices. You could use a for-of loop.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
https://google.github.io/styleguide/jsguide.html#features-for-loops